### PR TITLE
Make the changes necessary to use this gem in a Rails 5.2 context

### PR DESCRIPTION
**Pull request by zuver**
# Objective

Supports https://www.pivotaltracker.com/story/show/161830129, but doesn't complete it. There will be another PR in the `trakstar` repository that will integrate the work in this PR.

The purpose of this PR is to get `vestal_versions` ready for Rails 5.

# Notes

The trickiest part of this upgrade was navigating the under-the-hood `activerecord` changes from `4.2.10` to `5.2.1`. Simply using `bundle update activerecord` and troubleshooting from there made it _extremely difficult_ to see why the latest gems were breaking `vestal_versions` specs. The best strategy I could come up with was to set strategic breakpoints in both my `4.2.10` and `5.2.1` environments and compare the state of the in-memory data at various points of the spec suite. At one point, I even used the `#methods` method to compare the methods available on `activerecord` `4.2.10` models vs. `5.2.1` models 😂 that's how you really know debugging could be going better.

My initial brute force approach did help a little because I ultimately discovered a key overridden method that was not getting called in the `5.2.1` context -- a method called `changes` in the class `Version`. This method is simply there to override the default `activerecord` definition, but it wasn't getting called in my `5.2.1` bundle. That was when I realized the delta between the old and new versions of `activerecord` is fairly gnarly as far as `vestal_versions` is concerned.

I began reading more and more documentation and saw that there was some dramatic changes from `4.2.10` to `5.2.1`, so I had what ended up being a great idea -- incrementally update the gems. So I tried `5.0.x` and `5.1.x` and saw if I could get those upgrades to work. Using this strategy led me in the right direction because `5.1.x` happens to issue some very helpful deprecation warnings if you use a feature of `activerecord` that will soon be phased out.

I got this screenshot from someone's blog, but this is literally the same message I saw when I was in the `5.1.x` phase of the upgrade. This message was absolutely critical in learning how to update the code here to be Rails 5.2-ready:
screen shot 2018-11-14 at 9 14 39 am

Basically, I renamed _most_ calls to `changes` with a call to `saved_changes`.

There were a couple updates I made because of other deprecation warning I saw, but if I recall most of these were deprecation warnings for Rails 6.

# Testing

- [ ] `bundle install` and `bundle exec rspec spec` passes

The "real" test will be using this gem with the Trakstar `rails5` branch and seeing if versioning still works. Admittedly, I haven't done this yet but plan to very soon.

### CODE REVIEW

**Comment by zuver** [comment]
I just realized that it would be wiser to first test `5.1.6` since our current Trakstar `rails5` branch is on that version of Rails. So I just pushed a commit to lock this gem to use dependencies no greater than `5.1.6`.

However, when we are ready it should be easy enough to upgrade this gem to `5.2.1`. I've already tested that context as well 👍

**Review by serogers — APPROVED** [review]
Awesome job investigating this! The changes (no pun intended) look good and I appreciate you taking on the Rails 6 deprecation warnings. Based on my testing and the chat we had about the `saved_changes` method, I think these updates are good to go!
